src/kaboom.ts

```diff
@@ -2458,6 +2472,109 @@ function addLevel(map: string[], opt: LevelOpt): Level {
 
 }
 
+function addIsometricLevel(map: string[], options: IsometricLevelOpt): IsometricLevel {
+	if (!options.width || !options.height) {
+		throw new Error("Must provide isometric level tile width & height.");
+	}
+
+	const objects: GameObj[] = [];
+	const offset = vec2(options.pos || vec2(0));
+
+	const halfTileWidth = Math.floor(options.width / 2);
+	const halfTileHeight = Math.floor(options.height / 2);
+
+	const maxWidthInTiles = map.reduce((width, row): number => Math.max(width, row.length), 0)
+	const heightInTiles = map.length;
+
+	const isometricLevel = {
+		offset() {
+			return offset.clone();
+		},
+
+		gridWidth() {
+			return options.width;
+		},
+
+		gridHeight() {
+			return options.height;
+		},
+
+		fromIsometricCoordsToWorldPos: (row: number, col: number): Vec2 => {
+			return vec2((col - row) * halfTileWidth, (col + row) * halfTileHeight);
+		},
+
+		from2dTo1dIndex: (row: number, col: number): number => {
```

Comment:
When would this function be useful to user (there doesn't seem to be a way to index from this 1d index)?

Reply:
I thought this could enable the user to easily lookup the original level array to get the symbol for a specific tile, and from the symbol they can get its metadata in another object or something.

Is there a way to get which tile the user clicked on for example? I thought of the following use case but I'm not familiar with kaboomjs enough to say more yet:

1- player is sitting at tile X,Y
2- player clicks on tile W,Z
3- from W,Z tile coord you can get one dimensional array position (k) to lookup level declaration and get tile symbol
4- from tile symbol you can get some info elsewhere or just run some kind of pathfinding algorithm from X,Y (i) to W,Z (k).
5- walk along the generated path considering collision etc

Is there a better way to implement something like A* pathfinding algorithm in this case?

Sorry I haven't done much in kaboom yet so I'm not so familiar with the API yet, I'm waiting for the weekend to come x)